actors/miner/src/policy.rs

```diff
@@ -114,15 +123,36 @@ pub fn quality_for_weight(
     let scaled_up_weighted_sum_space_time: SectorQuality =
         weighted_sum_space_time << SECTOR_QUALITY_PRECISION;
 
-    scaled_up_weighted_sum_space_time
+    let quality = scaled_up_weighted_sum_space_time
         .div_floor(&sector_space_time)
-        .div_floor(&QUALITY_BASE_MULTIPLIER)
+        .div_floor(&QUALITY_BASE_MULTIPLIER);
+    // Apply SDM
+    sdm_scale(quality, duration)
+}
+
+/// Scale `value` by the sector duration multiplier for `duration`.
+pub fn sdm_scale<T>(mut value: T, duration: ChainEpoch) -> T
+where
+    T: MulAssign<ChainEpoch> + DivAssign<ChainEpoch>,
+{
+    if duration > SECTOR_DURATION_MULTIPLIER_START {
+        if duration > SECTOR_DURATION_MULTIPLIER_END {
+            log::error!("sector lifetime {duration} exceeded maximum expected lifetime {SECTOR_DURATION_MULTIPLIER_END}");
+            value *= 2;
+        } else {
+            let numerator = duration - SECTOR_DURATION_MULTIPLIER_START;
+            let denominator = SECTOR_DURATION_MULTIPLIER_END - SECTOR_DURATION_MULTIPLIER_START;
+            // scale from 1x to 2x.
```

Comment:
Is the `2x` here entirely implicit? There's no constant for slope. Even if it can be written this way for a max multiplier of 2, I think we should instead write it to transparently use a parameter for the max multiplier.

Reply:
It was a bit nasty when I last tried that, but I can see how it turns out.

Reply:
Eh, nvm. Turned out fine.